filebeat/module/elasticsearch/audit/_meta/fields.yml

```diff
@@ -22,6 +22,14 @@
       description: "The principal (username) that failed authentication"
       example: "_anonymous"
       type: keyword
+    - name: realm
+      description: "The authentication realm"
+      example": "active_directory"
+      type: keyword
+    - name: roles
+      description: "Roles to which the principal belongs"
+      example: [ "kibana_user", "beats_admin" ]
+      type: array
```

Comment:
This should be type: keyword ?

Reply:
I was wondering about this myself. I went with `array` because I saw it used in a couple other places, e.g.:

- https://github.com/elastic/beats/blob/6.x/filebeat/module/nginx/access/_meta/fields.yml#L7
- https://github.com/elastic/beats/blob/6.x/metricbeat/module/docker/container/_meta/fields.yml#L37
- https://github.com/elastic/beats/blob/6.x/packetbeat/protos/memcache/_meta/fields.yml#L105

I realize Elasticsearch/Lucene doesn't care about arrays; every field can be an array. So when is it appropriate to use `array` vs. the primitive data type in `fields.yml`?

Reply:
I think the usage in these places is actually a bug. If I remember correctly we reserved the `array` usage in the old days when there was an array of objects (nested objects). Probably we should rename this ...

For the other 3 issue, could you open an issue or directly a PR to fix it?

Reply:
Thanks for clarifying. I will fix the data types in this PR's `fields.yml` to be primitives and make two additional PRs: one to `master` to fix this PR's mistake in `master` and one to fix the other 3 usages (and many others, in fact).

Reply:
thx

Reply:
https://github.com/elastic/beats/pull/10307